api/services/get.js

```diff
@@ -0,0 +1,152 @@
+    _get.bind(null, bag),
+    _getServicesJson.bind(null, bag),
+    _setDefault.bind(null, bag)
+  ],
+    function (err) {
+      logger.info(bag.who, 'Completed');
+      if (err)
+        return respondWithError(res, err);
+
+      sendJSONResponse(res, bag.resBody);
+    }
+  );
+}
+
+function _checkInputParams(bag, next) {
+  var who = bag.who + '|' + _checkInputParams.name;
+  logger.verbose(who, 'Inside');
+
+  if (bag.reqQuery.name)
+    bag.serviceQuery = bag.reqQuery.name;
+
+  return next();
+}
+
+function _get(bag, next) {
+  var who = bag.who + '|' + _get.name;
+  logger.verbose(who, 'Inside');
+
+  configHandler.get(bag.component,
+    function (err, services) {
+      if (err)
+        return next(
+          new ActErr(who, ActErr.DBOperationFailed,
+            'Failed to get ' + bag.component, err)
+        );
+
+      if (_.isEmpty(services)) {
+        logger.debug('No configuration in database for ' + bag.component);
+        bag.initializeDefault = true;
+        return next();
+      }
+
+      if (bag.serviceQuery) {
+        if (!services[bag.serviceQuery])
+          return next(
+            new ActErr(who, ActErr.DataNotFound,
+              util.format('No service \'%s\' found in systemSettings.services',
+                bag.serviceQuery), err)
+          );
+        bag.resBody = services[bag.serviceQuery];
+
+      } else {
+        bag.resBody = services;
+      }
+
+      return next();
+    }
+  );
+}
+
+function _getServicesJson(bag, next) {
+  if (!bag.initializeDefault) return next();
+
+  var who = bag.who + '|' + _getServicesJson.name;
+  logger.verbose(who, 'Inside');
+
+  var servicesJsonPath =
+    path.join(global.config.scriptsDir, '/configs/services.json');
+
+  fs.readFile(servicesJsonPath,
+    function (err, data) {
+      if (err)
+        return next(
+          new ActErr(who, ActErr.OperationFailed,
+            'Failed to get services.json: ', err)
+        );
+
+      var error;
+
+      try {
+        bag.services = JSON.parse(data);
+      } catch (err) {
+        if (err)
+          error = new ActErr(who, ActErr.OperationFailed,
+            util.format('Failed to parse services.json: %s', err)
+          );
+      }
+
+      return next(error);
+    }
+  );
+}
+
+function _setDefault(bag, next) {
+  if (!bag.initializeDefault) return next();
+
+  var who = bag.who + '|' + _setDefault.name;
+  logger.verbose(who, 'Inside');
+
+  _.each(bag.services.serviceConfigs,
+    function (service) {
+      if (!service.isCore)
+        return;
+
+      bag.defaultService[service.name] = {
+        serviceName: service.name,
+        mounts: '',
+        envs: '',
+        opts: '',
+        replicas: service.isGlobal ? 'global' : 1
+      };
+    }
+  );
+
+  configHandler.put(bag.component, bag.defaultService,
+    function (err) {
+      if (err)
+        return next(
+          new ActErr(who, ActErr.OperationFailed, err)
+        );
+
+      bag.resBody = bag.defaultService;
```

Comment:
What if this route was called with a queryString and the default had not been set?